File: srutha/lib/models/video.dart
class Video {
  final String id;
  final String title;
  final String channelId;
  final String channelName;
  final String? description;
  final String? thumbnailUrl;
  final String url;
  final Duration? duration;
  final DateTime? uploadDate;
  final int? viewCount;

  Video({
    required this.id,
    required this.title,
    required this.channelId,
    required this.channelName,
    this.description,
    this.thumbnailUrl,
    required this.url,
    this.duration,
    this.uploadDate,
    this.viewCount,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'title': title,
      'channelId': channelId,
      'channelName': channelName,
      'description': description,
      'thumbnailUrl': thumbnailUrl,
      'url': url,
      'durationSeconds': duration?.inSeconds,
      'uploadDate': uploadDate?.toIso8601String(),
      'viewCount': viewCount,
    };
  }

  factory Video.fromMap(Map<String, dynamic> map) {
    return Video(
      id: map['id'] as String,
      title: map['title'] as String,
      channelId: map['channelId'] as String,
      channelName: map['channelName'] as String,
      description: map['description'] as String?,
      thumbnailUrl: map['thumbnailUrl'] as String?,
      url: map['url'] as String,
      duration: map['durationSeconds'] != null 
          ? Duration(seconds: map['durationSeconds'] as int) 
          : null,
      uploadDate: map['uploadDate'] != null 
          ? DateTime.parse(map['uploadDate'] as String) 
          : null,
      viewCount: map['viewCount'] as int?,
    );
  }

  String get formattedDuration {
    if (duration == null) return '';
    final hours = duration!.inHours;
    final minutes = duration!.inMinutes.remainder(60);
    final seconds = duration!.inSeconds.remainder(60);
    
    if (hours > 0) {
      return '$hours:${minutes.toString().padLeft(2, '0')}:${seconds.toString().padLeft(2, '0')}';
    }
    return '$minutes:${seconds.toString().padLeft(2, '0')}';
  }

  String get formattedViews {
    if (viewCount == null) return '';
    if (viewCount! >= 1000000) {
      return '${(viewCount! / 1000000).toStringAsFixed(1)}M views';
    } else if (viewCount! >= 1000) {
      return '${(viewCount! / 1000).toStringAsFixed(1)}K views';
    }
    return '$viewCount views';
  }
}

File: src/models/Video.ts
export interface Video {
  id: string;
  title: string;
  channelId: string;
  channelName: string;
  description?: string;
  thumbnailUrl?: string;
  url: string;
  durationSeconds?: number;
  uploadDate?: string;
  viewCount?: number;
  watched?: boolean;
  watchedDate?: string;
}

export interface VideoInput {
  id: string;
  title: string;
  channelId: string;
  channelName: string;
  description?: string;
  thumbnailUrl?: string;
  url: string;
  durationSeconds?: number;
  uploadDate?: string;
  viewCount?: number;
  watched?: boolean;
  watchedDate?: string;
}

// Helper functions for video formatting
export const formatDuration = (seconds?: number): string => {
  if (!seconds) return '';
  
  const hours = Math.floor(seconds / 3600);
  const minutes = Math.floor((seconds % 3600) / 60);
  const secs = seconds % 60;
  
  if (hours > 0) {
    return `${hours}:${minutes.toString().padStart(2, '0')}:${secs.toString().padStart(2, '0')}`;
  }
  return `${minutes}:${secs.toString().padStart(2, '0')}`;
};

export const formatViews = (views?: number): string => {
  if (!views) return '';
  
  if (views >= 1000000) {
    return `${(views / 1000000).toFixed(1)}M views`;
  } else if (views >= 1000) {
    return `${(views / 1000).toFixed(1)}K views`;
  }
  return `${views} views`;
};

export const getTimeAgo = (dateString: string): string => {
  const date = new Date(dateString);
  const now = new Date();
  const diffMs = now.getTime() - date.getTime();
  const diffSecs = Math.floor(diffMs / 1000);
  const diffMins = Math.floor(diffSecs / 60);
  const diffHours = Math.floor(diffMins / 60);
  const diffDays = Math.floor(diffHours / 24);
  const diffMonths = Math.floor(diffDays / 30);
  const diffYears = Math.floor(diffDays / 365);

  if (diffYears > 0) {
    return `${diffYears} ${diffYears === 1 ? 'year' : 'years'} ago`;
  } else if (diffMonths > 0) {
    return `${diffMonths} ${diffMonths === 1 ? 'month' : 'months'} ago`;
  } else if (diffDays > 0) {
    return `${diffDays} ${diffDays === 1 ? 'day' : 'days'} ago`;
  } else if (diffHours > 0) {
    return `${diffHours} ${diffHours === 1 ? 'hour' : 'hours'} ago`;
  } else if (diffMins > 0) {
    return `${diffMins} ${diffMins === 1 ? 'minute' : 'minutes'} ago`;
  } else {
    return 'Just now';
  }
};
